src/CMakeLists.txt

```diff
@@ -0,0 +1,95 @@
+find_package(Freetype REQUIRED)
+find_package(glad CONFIG REQUIRED)
+find_package(glm CONFIG REQUIRED)
+find_package(SDL2 CONFIG REQUIRED)
+find_package(sdl2-image CONFIG REQUIRED)
+find_package(sdl2-mixer CONFIG REQUIRED)
+find_package(spdlog CONFIG REQUIRED)
+
+# lua doesn't provide native cmake support
+find_path(LUA_INCLUDE_DIR lua.h)
+include_directories(${LUA_INCLUDE_DIR})
+
+find_library(LUA_LIBRARY lua)
+link_libraries(${LUA_LIBRARY})
+
+find_package(Doxygen QUIET)
+
+file(GLOB_RECURSE NOVELRT_LIB_HEADERS CONFIGURE_DEPENDS ../include/*.h)
+source_group(TREE ${CMAKE_SOURCE_DIR} FILES ${NOVELRT_LIB_HEADERS})
+
+file(GLOB_RECURSE NOVELRT_LIB_SOURCES CONFIGURE_DEPENDS NovelRT/*.cpp)
+source_group(TREE ${CMAKE_SOURCE_DIR} FILES ${NOVELRT_LIB_SOURCES})
+
+set(NOVELRT_LIB_LINK_LIBRARIES
+  Freetype::Freetype
+  glad::glad
+  glm
+  SDL2::SDL2-static
+  SDL2::SDL2_image
+  SDL2::SDL2_mixer
+  SDL2::SDL2main
+  spdlog::spdlog
+)
+
+if(NOT WIN32)
+  set(NOVELRT_LIB_LINK_LIBRARIES
+    ${NOVELRT_LIB_LINK_LIBRARIES}
+    m
+  )
+endif()
+
+if (WIN32)
+  set(CMAKE_WINDOWS_EXPORT_ALL_SYMBOLS ON)
+endif()
+
+add_library(libNovelRT SHARED ${NOVELRT_LIB_HEADERS} ${NOVELRT_LIB_SOURCES})
+target_link_libraries(libNovelRT ${NOVELRT_LIB_LINK_LIBRARIES})
+
+file(GLOB_RECURSE NOVELRT_LIB_FONT_RESOURCES CONFIGURE_DEPENDS Resources/Fonts/*.ttf)
+file(GLOB_RECURSE NOVELRT_LIB_IMAGE_RESOURCES CONFIGURE_DEPENDS Resources/Images/*.png)
+file(GLOB_RECURSE NOVELRT_LIB_SCRIPT_RESOURCES CONFIGURE_DEPENDS Resources/Scripts/*.lua)
+file(GLOB_RECURSE NOVELRT_LIB_SHADER_RESOURCES CONFIGURE_DEPENDS Resources/Shaders/*.glsl)
+
+add_custom_command(TARGET libNovelRT POST_BUILD
```

Comment:
Notably, rather than dumping all assets to the root output directory, this mirrors the directory structure (e.g. `<out>/Resources/Images/novel-chan.png`).

This helps organize the output and isolate assets from eachother.